**Objective of the notebook:**

주택 구매를 하기 위해 부동산을 알아볼 때, 해당 건물에 대한 다양한 정보들이 주택의 매매가에 영향을 미침을 알 수 있다.
이번에는 주택에 대한 79개의 Feature들을 통하여 Iowa주의 Ames에 있는 주택의 가격을 예측하고자 한다.




In [3]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

**About the dataset:**

House Prices: Advanced Regression Techniques

출처 : Kaggle (https://www.kaggle.com/c/house-prices-advanced-regression-techniques)

Kaggle에서는 다음과 같은 데이터를 제공한다.
* train.csv
* test.csv

데이터셋의 각 행은 하나의 주택의 정보와 가격이다.

데이터 필드는 다음과 같다.

In [9]:
variable = pd.read_excel('variable.xlsx')
variable

Variable        Scale  \
0              Id  Categorical   
1      MSSubClass  Categorical   
2        MSZoning  Categorical   
3     LotFrontage   Continuous   
4         LotArea   Continuous   
5          Street  Categorical   
6           Alley  Categorical   
7        LotShape  Categorical   
8     LandContour  Categorical   
9       Utilities  Categorical   
10      LotConfig  Categorical   
11      LandSlope  Categorical   
12   Neighborhood  Categorical   
13     Condition1  Categorical   
14     Condition2  Categorical   
15       BldgType  Categorical   
16     HouseStyle  Categorical   
17    OverallQual      Ordinal   
18    OverallCond      Ordinal   
19      YearBuilt   Continuous   
20   YearRemodAdd   Continuous   
21      RoofStyle  Categorical   
22       RoofMatl  Categorical   
23    Exterior1st  Categorical   
24    Exterior2nd  Categorical   
25     MasVnrType  Categorical   
26     MasVnrArea   Continuous   
27      ExterQual      Ordinal   
28      ExterCond      Ordinal   
29     Foundation  Categorical   
..            ...          ...   
51   BedroomAbvGr      Ordinal   
52   KitchenAbvGr      Ordinal   
53    KitchenQual      Ordinal   
54   TotRmsAbvGrd      Ordinal   
55     Functional  Categorical   
56     Fireplaces   Continuous   
57    FireplaceQu  Categorical   
58     GarageType  Categorical   
59    GarageYrBlt   Continuous   
60   GarageFinish  Categorical   
61     GarageCars   Continuous   
62     GarageArea   Continuous   
63     GarageQual  Categorical   
64     GarageCond  Categorical   
65     PavedDrive  Categorical   
66     WoodDeckSF   Continuous   
67    OpenPorchSF   Continuous   
68  EnclosedPorch   Continuous   
69      3SsnPorch   Continuous   
70    ScreenPorch   Continuous   
71       PoolArea   Continuous   
72         PoolQC      Ordinal   
73          Fence  Categorical   
74    MiscFeature  Categorical   
75        MiscVal      Ordinal   
76         MoSold  Categorical   
77         YrSold  Categorical   
78       SaleType  Categorical   
79  SaleCondition  Categorical   
80      SalePrice   Continuous   

                                              Meaning  \
0                                                  ID   
1   Identifies the type of dwelling involved in th...   
2   Identifies the general zoning classification o...   
3         Linear feet of street connected to property   
4                             Lot size in square feet   
5                     Type of road access to property   
6                    Type of alley access to property   
7                           General shape of property   
8                            Flatness of the property   
9                         Type of utilities available   
10                                  Lot configuration   
11                                  Slope of property   
12         Physical locations within Ames city limits   
13                    Proximity to various conditions   
14  Proximity to various conditions (if more than ...   
15                                   Type of dwelling   
16                                  Style of dwelling   
17  Rates the overall material and finish of the h...   
18           Rates the overall condition of the house   
19                         Original construction date   
20  Remodel date (same as construction date if no ...   
21                                       Type of roof   
22                                      Roof material   
23                         Exterior covering on house   
24  Exterior covering on house (if more than one m...   
25                                Masonry veneer type   
26                 Masonry veneer area in square feet   
27  Evaluates the quality of the material on the e...   
28  Evaluates the present condition of the materia...   
29                                 Type of foundation   
..                                                ...   
51                                Bedroom above grade   
52                 

In [23]:
train_df = pd.read_csv('train.csv')
test_df = pd.read_csv('test.csv')

In [24]:
print('Train Data Shape : {}'.format(train_df.shape))
print('Test Data Shape : {}'.format(test_df.shape))

Train Data Shape : (1460, 81)
Test Data Shape : (1459, 80)


Train 데이터에는 1460개의 데이터와 81개의 변수가 존재한다.
Test 데이터는 Kaggle competition에서 데이터를 가져왔기 떄문에 SalePrice가 존재하지 않는다.

In [21]:
train_df.head()

Id  MSSubClass MSZoning  LotFrontage  LotArea Street Alley LotShape  \
0   1          60       RL         65.0     8450   Pave   NaN      Reg   
1   2          20       RL         80.0     9600   Pave   NaN      Reg   
2   3          60       RL         68.0    11250   Pave   NaN      IR1   
3   4          70       RL         60.0     9550   Pave   NaN      IR1   
4   5          60       RL         84.0    14260   Pave   NaN      IR1   

  LandContour Utilities    ...     PoolArea PoolQC Fence MiscFeature MiscVal  \
0         Lvl    AllPub    ...            0    NaN   NaN         NaN       0   
1         Lvl    AllPub    ...            0    NaN   NaN         NaN       0   
2         Lvl    AllPub    ...            0    NaN   NaN         NaN       0   
3         Lvl    AllPub    ...            0    NaN   NaN         NaN       0   
4         Lvl    AllPub    ...            0    NaN   NaN         NaN       0   

  MoSold YrSold  SaleType  SaleCondition  SalePrice  
0      2   2008        WD         Normal     208500  
1      5   2007        WD         Normal     181500  
2      9   2008        WD         Normal     223500  
3      2   2006        WD        Abnorml     140000  
4     12   2008        WD         Normal     250000  

[5 rows x 81 columns]

In [19]:
train.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1460 entries, 0 to 1459
Data columns (total 81 columns):
Id               1460 non-null int64
MSSubClass       1460 non-null int64
MSZoning         1460 non-null object
LotFrontage      1201 non-null float64
LotArea          1460 non-null int64
Street           1460 non-null object
Alley            91 non-null object
LotShape         1460 non-null object
LandContour      1460 non-null object
Utilities        1460 non-null object
LotConfig        1460 non-null object
LandSlope        1460 non-null object
Neighborhood     1460 non-null object
Condition1       1460 non-null object
Condition2       1460 non-null object
BldgType         1460 non-null object
HouseStyle       1460 non-null object
OverallQual      1460 non-null int64
OverallCond      1460 non-null int64
YearBuilt        1460 non-null int64
YearRemodAdd     1460 non-null int64
RoofStyle        1460 non-null object
RoofMatl         1460 non-null object
Exterior1st      1460 non-n

**result from the seminar:**

위 데이터는 Feature에 해당하는 변수가 79개이다. 분석을 하기 위해서는 feature engineering을 통하여 Feature의 갯수를 줄일 필요가 있다. feature engineering에는 많은 시간이 소요될 것이라 예상되기에(아직 feature engineering에 대한 지식이 부족함으로 판단) 우선적으로 전처리에 시간에 크게 소요되지 않는 데이터를 분석 후 House Prices 데이터를 살펴보기로 하였다. 